src/key_scan.rs

```diff
@@ -372,11 +373,17 @@ pub fn receive_history_entry<L: Ledger>(
     };
 
     let txn_asset = last_record.asset_def.code;
-    TransactionHistoryEntry {
+    TransactionParams::<L> {
+        uid: Some(uid),
+        timeout: None,
+        status: TransactionStatus::Unknown,
```

Comment:
I think this should be `TransactionStatus::Retired` -- if we're receiving the transaction, it must have been finalized, right?